### ISSUE TITLE
coreWebView2Deferral.Complete() opens in Edge only when target="_blank" link tag.

### ISSUE THREAD

**Issue by hiroki-niinuma** [MicrosoftEdge/WebView2Feedback#881]
**Description**

Only when target="_blank", the new form's webview2 will not navigate, and when "deferral.Complete()", the new Edge will launch and open the link without triggering any events.

**Version**
SDK: 1.0.705.50  and 1.0.781-prerelease-
Runtime: 88.0.705.56 (Official build) (64-bit) 
Framework: .net framework 4.8 
OS:Win10 64bit  

**Repro Steps**

```
using System;
using System.Windows.Forms;

namespace WindowsFormsApp14
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        Microsoft.Web.WebView2.Core.CoreWebView2NewWindowRequestedEventArgs eventArgs = null;
        Microsoft.Web.WebView2.Core.CoreWebView2Deferral deferral = null;
        public Form1(Microsoft.Web.WebView2.Core.CoreWebView2NewWindowRequestedEventArgs e, Microsoft.Web.WebView2.Core.CoreWebView2Deferral _deferral)
            : this()
        {
            eventArgs = e;
            deferral = _deferral;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            webView21.EnsureCoreWebView2Async(null);
            webView21.CoreWebView2InitializationCompleted += WebView21_CoreWebView2InitializationCompleted;
        }

        private void WebView21_CoreWebView2InitializationCompleted(object sender, Microsoft.Web.WebView2.Core.CoreWebView2InitializationCompletedEventArgs e)
        {
            if (deferral != null)
            {
                eventArgs.NewWindow = webView21.CoreWebView2; 
                deferral.Complete(); /// A new Edge will open without navigation.
                return;
            }
            webView21.CoreWebView2.Navigate("about:blank");
            webView21.CoreWebView2.NavigationCompleted += CoreWebView2_NavigationCompleted;
            webView21.CoreWebView2.NewWindowRequested += CoreWebView2_NewWindowRequested; ;
        }

        private void CoreWebView2_NewWindowRequested(object sender, Microsoft.Web.WebView2.Core.CoreWebView2NewWindowRequestedEventArgs e)
        {
            var d = e.GetDeferral();
            var f = new Form1(e,d);
            f.Show();

        }

        private void CoreWebView2_NavigationCompleted(object sender, Microsoft.Web.WebView2.Core.CoreWebView2NavigationCompletedEventArgs e)
        {
            // GOOD
            // webView21.NavigateToString("<a href=\"https://example.com\" target=\"AAA\">sample</a>");
            // BAD
            webView21.NavigateToString("<a href=\"https://example.com\" target=\"_blank\">sample</a>");
            webView21.CoreWebView2.NavigationCompleted -= CoreWebView2_NavigationCompleted;
        }
    }
}

```
**Additional context**

thanks.

[AB#31577848](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/31577848)

**Comment by champnic** [comment]
Thanks for the report, I've added a bug to our backlog to take a closer look.

**Comment by Netxxl** [comment]
We do have the same or similar issue with the NewWindowRequested Event:

Clicking on a Link in original Form WebView we want any new windows to open in one second form.
Code about the same as above.
Works while holding down Shift Key while clicking. Without it opens a complete new out-of process Edge Window.
Please can you look into this, thanks.

**Comment by maurawinstanley** [comment]
We are looking into a fix for this issue, in the meantime a workaround is adding a rel=\"opener\" to the NavigateToString call so that the call looks like this:
webView21.NavigateToString("<a rel=\"opener\" href=\"https://example.com\" target=\"_blank\">sample</a>");

**Comment by kolozsaron** [comment]
I can confirm that rel="opener" workaround works. Even I could create a general workaround:
```cs
private void webView2_CoreWebView2InitializationCompleted( object sender, Microsoft.Web.WebView2.Core.CoreWebView2InitializationCompletedEventArgs e )
{
    webView2.CoreWebView2.AddScriptToExecuteOnDocumentCreatedAsync( @"window.onload = () => {var as = document.getElementsByTagName('a');
                                                                        for(var i=0;i<as.length;i++){ as[i].rel='opener'; } }" );
```
But unfortunately the [CoreWebView2.WebResourceRequested](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.webresourcerequested?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(Microsoft.Web.WebView2.Core.CoreWebView2.WebResourceRequested)%3Bk(TargetFrameworkMoniker-.NETFramework%2CVersion%253Dv4.7.2)%3Bk(DevLang-csharp)%26rd%3Dtrue&view=webview2-dotnet-1.0.774.44) event is **not** risen in the new window (the NavigationStarting & NavigationCompleted events work only). This is a showstopper for us, because our application has to filter out some web resources (e.g. specific PDF-s.)

If I open the links using "Open link in new window" context menu item, the WebResourceRequested event works correctly in the new window.

Our large problem is that some of our customer websites stopped working in our old WebBrowser control solution, but WebView2 is still not applicable because of this bug.

WebView2: 1.0.774.44
Edge WebView2 Runtime: 89.0.774.57

**Comment by darbid** [comment]
I have just noticed this one too. The links have a target="_blank".

In WebView2 I get a NewWindowRequested event and a new window, but then another window opens without any notification.

In Edge when I click the same link then whatever the navigation should do happens in the new tab.

**Comment by kolozsaron** [comment]
Also I realized, that even these buttons are opened in new WebView2 window correctly:

```html
        <button onclick="window.open('https://www.microsoft.com')">test3</button>
        <form action="https://www.microsoft.com" target="_blank">
            <input type="submit" value="test4"/>
        </form>
```

and while NavigationStarting & NavigationCompleted events are risen for them as expected, the [CoreWebView2.WebResourceRequested](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.webresourcerequested?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(Microsoft.Web.WebView2.Core.CoreWebView2.WebResourceRequested)%3Bk(TargetFrameworkMoniker-.NETFramework%2CVersion%253Dv4.7.2)%3Bk(DevLang-csharp)%26rd%3Dtrue&view=webview2-dotnet-1.0.774.44) event is not risen for the new WebView2 window.

As I wrote in my [previous comment](https://github.com/MicrosoftEdge/WebView2Feedback/issues/881#issuecomment-804811793), this is an important issue for us as well. 
I don't know if this issue is related to this bug, or a completely new one.

**Comment by maurawinstanley** [comment]
We have fixed the deferral with target=_blank issue and the fix should be in all runtimes 93.0.924.0+. The fix will ship with the 93 version of the stable WebView2 Runtime scheduled for late July, thanks all! 

@kolozsaron regarding your issue, the new window must be set, and then if you listen for the webresourcerequested event once, it should fire - if you need an example please look at monica-ch's example in [this thread](https://github.com/MicrosoftEdge/WebView2Feedback/issues/1116).  I am going to go ahead and close this issue.
